### PR TITLE
Only allow arguments that have been registered to be set on the Request object

### PR DESCRIPTION
**Pull request by joehoyle**
See #1223 - this is an experiment, don't merge yet :)

### CODE REVIEW

**Comment by joehoyle** [comment]
@WP-API/amigos feedback appreciated here :)

File: lib/endpoints/class-wp-rest-comments-controller.php [hunk 1]
```diff
@@ -855,6 +855,12 @@ public function get_collection_params() {
 			'sanitize_callback' => 'absint',
 			'type'              => 'integer',
 		);
+		$query_params['context']   = array(
+			'default'           => 'view',
+			'description'       => 'Set the context for the requet. "view" or "edit".',
```
Comment (danielbachhuber) [inline lib/endpoints/class-wp-rest-comments-controller.php]: Spelling

Comment (danielbachhuber) [inline lib/endpoints/class-wp-rest-comments-controller.php]: Also, shouldn't the options be present as `enum`?

**Comment by danielbachhuber** [comment]
I am :100: to the idea.

**Comment by danielbachhuber** [comment]
To @rmccue for #reviewmerge

**Comment by rmccue** [comment]
@joehoyle and I talked this one over, and there's some conceptual things here that we both want to continue thinking over. Will compile coherent thoughts on it, but in the meantime, we decided to punt this one to b3.

**Comment by rmccue** [comment]
We're going to tone this one down a bit. Instead of requiring all arguments to be registered, we're going to require all registered arguments to have either a validation callback or a sanitisation callback (or both!). This stops people from accidentally misspelling `validate_callback`/`sanitize_callback`, but won't stop them from accessing other unregistered arguments.

This isn't a full solution that solves every problem, but we need to balance extensibility and usability here. We need people to use the request object as much as possible, and anything that encourages them to directly use `$_GET`/`$_POST` is going to hurt internal reuse.

Removed Review tag, and reassigning back to @joehoyle to switch it up.

**Comment by danielbachhuber** [comment]
> We're going to tone this one down a bit. Instead of requiring all arguments to be registered, we're going to require all registered arguments to have either a validation callback or a sanitisation callback (or both!). This stops people from accidentally misspelling validate_callback/sanitize_callback, but won't stop them from accessing other unregistered arguments.

Given the infrastructure has been committed to core, this still needs to be addressed: https://core.trac.wordpress.org/ticket/34725

**Pull request closed without merging**